### ISSUE TITLE
`xml2` error with current version of `kableExtra`

### ISSUE THREAD

**Issue by CeresBarros** [haozhu233/kableExtra#812]
I'm getting "Error in xml2::xml_length(thead) : Unexpected node type" with the current version of `kableExtra` but not with v1.3.4

Here's a reprex to the error.

```
install.packages(c("data.table", "kableExtra"), dependencies = TRUE)

library(data.table)
library(kableExtra)


profileTab <- data.table(nameShort = c("name1", "name2", "name3"),
                         photo = "",
                         nameDesc = "")

## add photos and descriptions -- not ideal, but Visual screws up the URLs
profileTab[nameShort == "name1", 
           photo := "![photo1](URL1){width='350'}"]
profileTab[nameShort == "name1", 
           nameDesc := paste("some text",
                             "[link](URL2)",
                             paste(
                               "more text"
                             ),
                             sep = "<br>")]
profileTab[nameShort == "name2", 
           photo := "![photo2](URL3){width='350'}"]
profileTab[nameShort == "name2", 
           nameDesc := paste("some text",
                             "[link](URL4)",
                             paste(
                               "more text"
                             ),
                             sep = "<br>")]

profileTab[nameShort == "name3", 
           photo := "![photo3](URL5){width='350'}"]
profileTab[nameShort == "name3", 
           nameDesc := paste("some text",
                             "[link](URL5)",
                             paste(
                               "more text"
                             ),
                             sep = "<br>")]

kable(profileTab[, .(photo, nameDesc)], format = "html", escape = FALSE, col.names = NULL) |> 
  kable_styling(full_width=TRUE) |>
  collapse_rows(columns = c(1,2), valign="top") |>
  column_spec(column = 1, border_right = TRUE, width = "4cm") |>
  print()
  ```

**Comment by dmurdoch** [comment]
The error happens in `kableExtra:::read_table_data_from_xml` because the table it receives has no `thead` part.  That function is called from `collapse_rows_html`, called from `collapse_rows`.  It has no `thead` part because you specified `col.names = NULL`.

Everything you did is legal, so this is a bug.  Looks pretty easy to fix, I'll give it a try.

**Comment by dmurdoch** [comment]
@haozhu233:  I forgot about that PR.  The fix I have just put together only applies to `kableExtra:::read_table_data_from_xml` so it's not as general as that one; on the other hand, I think it does a better job in this particular case.  (They do conflict.)  So I think I'll commit this narrow fix, then look into fixing up #790 to be compatible.

**Comment by CeresBarros** [comment]
thanks for the swift fix!
